### PR TITLE
Allow for reporting of loss curve from KerasModel.fit

### PR DESCRIPTION
**Pull request by rbharath**
This PR allows for the return of the full loss curve from `KerasModel.fit` by adding a new argument `return_loss_curve` as discussed in https://github.com/deepchem/deepchem/issues/2058 and adds a unit test.

This PR also fixes the logging issues from https://github.com/deepchem/deepchem/issues/1944.

### CODE REVIEW

**Comment by rbharath** [comment]
CC @peastman

**Comment by peastman** [comment]
It's a little ugly how this forces us to change the API of the base Model class, and also how we have to tell mypy to ignore errors because it can't infer the return type.  As an alternative, what if we added an argument `all_losses: Optional[List] = None`?  If you provide a list, it stores the losses into it.  A nice feature of this is that if you call `fit()` multiple times to continue training, you can keep passing in the same list, and values will keep getting appended to it.

**Comment by rbharath** [comment]
Good point! Let me try reworking this as you suggest

**Comment by rbharath** [comment]
This is much cleaner! No changes to API needed and solves our needs

File: deepchem/models/keras_model.py [hunk 1]
```diff
@@ -302,16 +303,20 @@ def fit(self,
     callbacks: function or list of functions
       one or more functions of the form f(model, step) that will be invoked after
       every step.  This can be used to perform validation, logging, etc.
+    all_losses: list, optional (default False)
```
Comment (peastman) [inline deepchem/models/keras_model.py:306]: Should be "default None".  Also we can give a more specific type: `Optional[List[float]]`.

File: deepchem/models/keras_model.py [hunk 2]
```diff
@@ -421,13 +433,19 @@ def fit_generator(self,
       avg_loss = float(avg_loss) / averaged_batches
       logger.info(
           'Ending global_step %d: Average loss %g' % (current_step, avg_loss))
+      avg_losses.append(avg_loss)
```
Comment (peastman) [inline deepchem/models/keras_model.py:436]: The implementation is a bit more complicated than it really needs to be.  The only required change to this method is a single addition here:

```python
if all_losses is not None:
  all_losses.append(avg_loss)
```

Everything else can stay exactly as it was before.

**Comment by peastman** [comment]
That looks much better!  I made a few additional minor suggestions.  Otherwise, I think it looks good.

File: deepchem/models/keras_model.py (hunk 1 above)
Comment (rbharath) [reply deepchem/models/keras_model.py:306]: Good suggestion, will fix!

File: deepchem/models/keras_model.py (hunk 2 above)
Comment (rbharath) [reply deepchem/models/keras_model.py]: Good suggestion! I've simplified it in this fashion, with a minor tweak to handle the edge case that was failing in #1944.

**Comment by rbharath** [comment]
Tests are green so going to go ahead and merge this in now!
